File: 프로그래머스/LV2/코딩테스트 연습/완전탐색/모음 사전/모음 사전.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <math.h>
int previous_count(char n)
{
    switch (n)
    {
    case 'A':
        return 0;
    case 'E':
        return 1;
    case 'I':
        return 2;
    case 'O':
        return 3;
    case 'U':
        return 4;    
    }
}

// 파라미터로 주어지는 문자열은 const로 주어집니다. 변경하려면 문자열을 복사해서 사용하세요.
int solution(const char* word) 
{
    int answer = 0;
    int vowel_pos = 0;
    for(int i = 0; i < strlen(word); i++)
    {
        answer += ((5 * (pow(5, 4 - i) - 1) / 4) + 1) * previous_count(word[i]) + 1;
    }
    return answer;
}

File: 프로그래머스/LV2/코딩테스트 연습/완전탐색/모음 사전/모음 사전.py
previous_count = {'A' : 0, 'E' : 1, 'I' : 2, 'O' : 3, 'U' : 4}

def solution(word):
    answer = 0
    for i, n in enumerate(word):
        answer += ((5 * (5**(4-i) - 1) // 4) + 1) * previous_count[n] + 1
    return answer

# vowel = ['A', 'E', 'I', 'O', 'U']
# answer = 0
# fin = False

# def go(select, s, word):
#     global answer, fin

#     if fin:
#         return
#     if s == word:
#         fin = True
#         return

#     answer += 1

#     if select == 5:
#         return

#     for v in vowel:
#         go(select + 1,  s + v, word)

# def solution(word):
#     global answer
#     go(0, "", word)
#     return answer
